charon/src/ast/values.rs

```diff
@@ -81,3 +82,30 @@ pub enum ScalarValue {
     U64(u64),
     U128(u128),
 }
+
+/// This is simlar to the Scalar value above. However, instead of storing
+/// the float value itself, we store its bits representation using a u128.
+/// This allows to derive the Eq and Ord traits, which are not implemented
+/// for floats
+#[derive(
+    Debug,
+    PartialEq,
+    Eq,
+    Copy,
+    Clone,
+    EnumIsA,
+    EnumAsGetters,
+    VariantName,
+    VariantIndexArity,
+    Hash,
+    PartialOrd,
+    Ord,
+    Drive,
+    DriveMut,
+)]
+pub enum FloatValue {
+    F16(u128),
+    F32(u128),
+    F64(u128),
+    F128(u128),
+}
```

Comment:
Instead of an enum, could this be
```rust
struct FloatValue {
    bits: u128,
    ty: FloatTy,
}
```
That's more convenient to manipulate most of the time.

Reply:
Do we even need the bit-value? We could store a string representation

Reply:
Done in https://github.com/AeneasVerif/charon/pull/348/commits/7e49300f449f6cc9bf27d6f09f756d07f5c0083b